From 4503a34ee38190b4aca89432951c0d100494d704 Mon Sep 17 00:00:00 2001
From: Pete Beardmore <pete.beardmore@msn.com>
Date: Sat, 18 Aug 2018 08:03:10 +0100
Subject: [mod] hackerrank, 'minimum absolute difference in an array', python
 solution

-speed-up, sort array then just iterate adjacent pairs
---
 .../minimum.absolute.difference.in.an.array.py           | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/hackerrank/interview.preparation.kit.-.greedy.algorithms.-.easy.-.minimum.absolute.difference.in.an.array/minimum.absolute.difference.in.an.array.py b/hackerrank/interview.preparation.kit.-.greedy.algorithms.-.easy.-.minimum.absolute.difference.in.an.array/minimum.absolute.difference.in.an.array.py
index 5a1f9fd..d9cfba6 100644
--- a/hackerrank/interview.preparation.kit.-.greedy.algorithms.-.easy.-.minimum.absolute.difference.in.an.array/minimum.absolute.difference.in.an.array.py
+++ b/hackerrank/interview.preparation.kit.-.greedy.algorithms.-.easy.-.minimum.absolute.difference.in.an.array/minimum.absolute.difference.in.an.array.py
@@ -8,10 +8,10 @@ import sys
 
 # Complete the minimumAbsoluteDifference function below.
 def minimumAbsoluteDifference(arr):
-    res = arr[1] - arr[0]
-    for idx in range(len(arr)):
-        for idx2 in range(idx + 1, len(arr)):
-            res = min([res, abs(arr[idx2] - arr[idx])])
+    arr.sort()
+    res = abs(arr[1] - arr[0])
+    for idx in range(len(arr) - 1):
+        res = min([res, abs(arr[idx + 1] - arr[idx])])
     return res
 
 if __name__ == '__main__':
@@ -26,4 +26,3 @@ if __name__ == '__main__':
     fptr.write(str(result) + '\n')
 
     fptr.close()
-
-- 
2.23.0

